### ISSUE TITLE
Add a S->C request/notification to update the clients config

### ISSUE THREAD

**Issue by Jamalam360** [microsoft/language-server-protocol#1692]
## Example

```jsonc
{
  // scope
  "myLsp": {
    "ignoredIdentifiers": [
       // existing content...
       "file"
     ]
  }
}
```

This would update the clients `myLsp.ignoredIdentifiers` configuration.

## Use Case

Take the following example:

```
file = { statement* }
statement = { "hello" | "goodbye" ~ ";" }
```

Our LSP for [Pest](https://github.com/pest-parser/pest-ide-tools) would report `file` as an unused rule, as it is only used from Rust code (which our server has no knowledge of). This is why we have the `alwaysUsedRuleNames` config, which suppresses that warning.

We want to provide a quick fix to add `file` to that list, but as far as I can tell there is no way for the server to request an update to the client's configuration file.

Is this against the principles of the LSP or just something that hasn't been implemented?

**Comment by dbaeumer** [comment]
In general I try to say away from write API (except a descriptive one like workspace / text edits) due to the async nature of the server. However I will keep it as a feature request.

**Comment by Jamalam360** [comment]
Yes fair enough. I just think this is a valid use case, so hopefully it will be looked at at some point :)
